homeassistant/components/ecobee/sensor.py

```diff
@@ -66,7 +94,7 @@ def __init__(
         self.sensor_name = sensor_name
         self.index = sensor_index
         self._state = None
-
+        self.thermostat = self.data.ecobee.get_thermostat(self.index)
```

Comment:
I don't believe we can cache the data as an instance attribute. When the ecobee client updates it replaces the object that holds all the thermostats. So after each update we need to look up new thermostat data from the client.

https://github.com/nkgilley/python-ecobee-api/blob/4878f97ba455ceae07739d44f77183fe75a00095/pyecobee/__init__.py#L202

Reply:
Yeah, that broke the updating. Need to call get_thermostats()